src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -169,20 +173,31 @@ public async Task UpdatePrimaryBranchSolutionAsync(AssetProvider assetProvider,
             }
         }
 
-        private Solution? TryGetAvailableSolution(Checksum solutionChecksum)
+        private Solution? TryGetAvailableSolution(Checksum solutionChecksum, ProjectId? projectId)
         {
```

Comment:
Keep the filter - it will preserve stack in case there is a bug in the above code

Reply:
i moved the filter to the higher frame.  is that not ok?

Reply:
Defeats the purpose - the filter needs to intercept the exception before the compiler-generated catch catches it in the async method.

Reply:
ah. :(